common/src/main/java/com/microsoft/identity/common/nativeauth/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -670,35 +668,31 @@ class NativeAuthMsalController : BaseNativeAuthController() {
         // Validate original AcquireTokenNoScopesCommandParameters parameters
         parameters.validate()
 
-        // Convert AcquireTokenNoScopesCommandParameters into SilentTokenCommandParameters,
-        // so we can use it in BaseController.getCachedAccountRecord()
-        val silentTokenCommandParameters =
-            CommandUtil.convertAcquireTokenNoFixedScopesCommandParameters(
-                parameters,
-                parameters.getCorrelationId()
-            )
+        // Add default scopes
+        val mergedScopes = addDefaultScopes(parameters)
+
+        val parametersWithScopes = parameters
```

Comment:
SDK result object's correlation ID is null, both in local-validation error scenarios (e.g. passing an empty scope list), as well as API error scenarios (e.g. passing an invalid scope). With success responses the correlation ID seems to be set.

![image](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/assets/1315628/db4a1383-02b0-4bd8-957d-204f4a4c4b26)

Reply:
I have set the correlation id to the value from accountstate and added tests for it in msal repo.

Reply:
I don't think that's correct. getAccessToken() is a new flow, and for every new flow we should be setting a new correlation ID.

Reply:
In the old code (https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/2082/files#diff-1b65b25a41283439e1932bae0a4dc86a788588813709114e41087ffde8b4ca09 AccountState.kt file line no 270), the correlation id for the flow is set to the correlation id of the accountstate object. If we want a new correlation id then it would be change in behaviour for the old getAccessToken() method.

Reply:
If we want to change the old behaviour and go with a new correlation id then the empty scope error should return "UNSET" as the correlation id and the invalid scope should return a proper UUID.

Reply:
I see. In that case let's leave it as it is right now. We can always change the behaviour at a later point in time.